File: Chapter01/Python_Code/UseRegularExpressionToFilterBadEmailAddresses.py
import os
import pandas as pd 

goodemails_raw = pd.read_csv("./Data/EmailAddresses.csv")
email_pattern = "^([a-zA-Z][\\w\\_]{4,20})\\@([a-zA-Z0-9.-]+)\\.([a-zA-Z]{2,3})$"
goodemails = goodemails_raw[goodemails_raw["Email"].str.match(email_pattern)]

File: Chapter03/R_Code/UseRegularExpressionToFilterBadEmailAddresses.R
library(tidyverse)
library(stringr)

setwd("<path to the R_Code folder>")
goodemails_raw <- read_csv("./Data/EmailAddresses.csv")

email_pattern <- "^([a-zA-Z][\\w\\_]{4,20})\\@([a-zA-Z0-9.-]+)\\.([a-zA-Z]{2,3})$"

goodemails <-
	goodemails_raw %>%
	filter(str_detect(Email, email_pattern) == TRUE)
